discovery/consul/consul_test.go

```diff
@@ -297,3 +297,89 @@ func TestAllOptions(t *testing.T) {
 	checkOneTarget(t, <-ch)
 	cancel()
 }
+
+func TestGetDatacenter_ShouldUseAgentInfo(t *testing.T) {
+	stub, config := newServer(t)
+	defer stub.Close()
+
+	d := newDiscovery(t, config)
+
+	// should be empty if not initialized
+	testutil.Equals(t, "", d.clientDatacenter)
+
+	err := d.getDatacenter()
+
+	// no error expected here
+	testutil.Equals(t, nil, err)
+	// should be value of agent info once initialized
+	testutil.Equals(t, "test-dc", d.clientDatacenter)
+
+	// second call should use value cached
+	err = d.getDatacenter()
+
+	// no error expected here
+	testutil.Equals(t, nil, err)
+	// should be value of agent info once initialized
+	testutil.Equals(t, "test-dc", d.clientDatacenter)
+}
+
+func TestGetDatacenter_ShouldReturnErrorGivenInternalServerError(t *testing.T) {
```

Comment:
You can use table-driven tests to group this test and the next one in a single test since most of the code is common.